### PR TITLE
[IM] Remove the EMBER_ZCL_STATUS_DUPLICATE_EXISTS error code

### PR DESCRIPTION
**Pull request by jepenven-silabs**
With the EmberAfStatus being replaced with the InteractionModel::Status, this error code won't exist anymore. Add a CHIP_ERROR instead (easier than modifying the spec for the InteractionModel::Status)

### CODE REVIEW

File: src/app/app-platform/ContentAppPlatform.cpp [hunk 1]
```diff
@@ -151,9 +151,9 @@ EndpointId ContentAppPlatform::AddContentApp(ContentApp * app, EmberAfEndpointTy
                 IncrementCurrentEndpointID();
                 return app->GetEndpointId();
             }
-            else if (ret != EMBER_ZCL_STATUS_DUPLICATE_EXISTS)
+            else if (err != CHIP_ERROR_ENDPOINT_EXISTS)
             {
-                ChipLogError(DeviceLayer, "Adding ContentApp error=%d", ret);
+                ChipLogError(DeviceLayer, "Adding ContentApp error=%s", ErrorStr(err));
```
Comment (bzbarsky-apple) [inline src/app/app-platform/ContentAppPlatform.cpp:156]:
```suggestion
                ChipLogError(DeviceLayer, "Adding ContentApp error=%" CHIP_ERROR_FORMAT, err.Format());
```

File: src/app/app-platform/ContentAppPlatform.cpp [hunk 2]
```diff
@@ -202,10 +202,10 @@ EndpointId ContentAppPlatform::AddContentApp(ContentApp * app, EmberAfEndpointTy
             index++;
             continue;
         }
-        EmberAfStatus ret = emberAfSetDynamicEndpoint(index, desiredEndpointId, ep, dataVersionStorage, deviceTypeList);
-        if (ret != EMBER_ZCL_STATUS_SUCCESS)
+        CHIP_ERROR err = emberAfSetDynamicEndpoint(index, desiredEndpointId, ep, dataVersionStorage, deviceTypeList);
+        if (err != EMBER_ZCL_STATUS_SUCCESS)
         {
-            ChipLogError(DeviceLayer, "Adding ContentApp error=%d", ret);
+            ChipLogError(DeviceLayer, "Adding ContentApp error : %s", ErrorStr(err));
```
Comment (bzbarsky-apple) [inline src/app/app-platform/ContentAppPlatform.cpp:208]:
```suggestion
            ChipLogError(DeviceLayer, "Adding ContentApp error : %" CHIP_ERROR_FORMAT, err.Format());
```

File: src/app/util/attribute-storage.h [hunk 3]
```diff
@@ -190,10 +190,10 @@ CHIP_ERROR SetTagList(chip::EndpointId endpoint,
 //
 // An optional parent endpoint id should be passed for child endpoints of composed device.
 //
-EmberAfStatus emberAfSetDynamicEndpoint(uint16_t index, chip::EndpointId id, const EmberAfEndpointType * ep,
-                                        const chip::Span<chip::DataVersion> & dataVersionStorage,
-                                        chip::Span<const EmberAfDeviceType> deviceTypeList = {},
-                                        chip::EndpointId parentEndpointId                  = chip::kInvalidEndpointId);
+CHIP_ERROR emberAfSetDynamicEndpoint(uint16_t index, chip::EndpointId id, const EmberAfEndpointType * ep,
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.h:193]: Should document the error cases and what they will return.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/util/attribute-storage.h (hunk 3 above)
Comment (jepenven-silabs) [reply src/app/util/attribute-storage.h:198]: done

File: docs/ERROR_CODES.md [hunk 4]
```diff
@@ -102,6 +102,7 @@ This file was **AUTOMATICALLY** generated by
 | 121       | 0x79  | `CHIP_ERROR_INSUFFICIENT_PRIVILEGE`                |
 | 125       | 0x7D  | `CHIP_ERROR_MESSAGE_COUNTER_EXHAUSTED`             |
 | 126       | 0x7E  | `CHIP_ERROR_FABRIC_EXISTS`                         |
+| 127       | 0x7F  | `CHIP_ERROR_ENDPOINT_EXISTS`                       |
```
Comment (jmartinez-silabs) [inline docs/ERROR_CODES.md:105]: Was this and old deprecated error code we can reuses or should we use a new value?

**Review by jmartinez-silabs — APPROVED** [review]

File: docs/ERROR_CODES.md (hunk 4 above)
Comment (jepenven-silabs) [reply docs/ERROR_CODES.md:105]: It was a free available error code. unless the code comments were wrong.

**Pull request merged**